lib/ash_authentication/strategies/password.ex

```diff
@@ -94,7 +94,8 @@ defmodule AshAuthentication.Strategy.Password do
   See the [Testing guide](/documentation/topics/testing.md) for tips on testing resources using this strategy.
   """
 
-  defstruct confirmation_required?: false,
+  defstruct require_confirmed_with?: false,
```

Comment:
The option should not end in a `?`, that is only for boolean options. This should be `nil | atom`